### ISSUE TITLE
postgres source: inconsistent behavior of `__timeFilter` macro in alert manager rule

### ISSUE THREAD

**Issue by ttys3** [grafana/grafana#70786]
### 
# What went wrong?

**What happened**:

the query workds good under **Explore** and grafana panel:

```sql
select timestamp as time, count(*) as error_count from request_total 
where error > 0 AND $__timeFilter(timestamp)
```
, it generated the right query condition

 `timestamp BETWEEN '2023-06-28T03:19:22.473Z' AND '2023-06-28T06:19:22.473Z'`

but it failed in edit alert rule "Set a query and alert condition"

because it generated something like that 

` '2023-06-28T03:19:34.410235195Z' and '2023-06-28T06:19:34.410235195Z'`

this breaks the query in QuestDB

the same issue exists for  `__timeFrom` and `__timeTo` 

**What did you expect to happen**:

`__timeFilter`,  `__timeFrom` and `__timeTo` behavior  in alert rule should be consistent with panel and explorer

### 
# How do we reproduce it?

**Step 1**:

- Open Grafana and do X 

**Step 2**:

- Now click button Y

**Step 3**:

- Wait for the browser to crash. Error message says: "Error..."

### 
# What Grafana version are you using?

v10.0

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

Docker

### User's OS?

Fedora 38

### User's Browser?

Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.58

### Is this a Regression?

None

### Are Datasources involved?

QuestDB 7.2 with postgres source in grafana

**Comment by usmangt** [comment]
Hi @ttys3,

Thanks for opening this issue.

I will forward this to our Engineers so that they can have a look and might be able to tell more details about it.

**Comment by armandgrillet** [comment]
We have not been able to reproduce this in Grafana with Postgresql. From the QuestDB docs: "The native timestamp format used by QuestDB is a Unix timestamp in microsecond resolution. Although timestamps in nanoseconds will be parsed, the output will be truncated to microseconds." which could be an issue.

We don't have enough details from the current issue, could you send us screenshots of how it looks in the dashboard and the alert rule creation form?

**Comment by rwwiv** [comment]
This happens because Explore generates the `from` and `to` ranges for the query in the browser using a Javascript `DateTime`, which has only millisecond accuracy. Alerting generates those timestamps on the server, and Go creates times with nanosecond accuracy. 

The code that actually generates that `BETWEEN <time> AND <time>` ([see here](https://github.com/grafana/grafana/blob/3ee26df41ec82e9fafdf98453b4047e5ecb0e61c/pkg/tsdb/postgres/macros.go#L87)) will format the time using nanosecond accuracy, but does not pad the time if less than nanosecond accuracy is passed in. 

This behavior is confusing and ideally this would be consistent across the app. However, given that QuestDB's docs state that it should parse nanosecond timestamps [here](https://questdb.io/docs/guides/working-with-timestamps-timezones/#timestamps-in-questdb) this does seem to be a bug in QuestDB.

**Comment by gillesdemey** [comment]
This has been fixed in QuestDB 7.3.1 https://github.com/questdb/questdb/pull/3626
